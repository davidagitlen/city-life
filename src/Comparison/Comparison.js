import React from 'react';
import './Comparison.scss';
import { connect } from 'react-redux';


const Comparison = (props) => {
  console.log(props)
  return(
    <div className='Comparison'>
      <img src='' alt=''/>
      <p><span>Full Name:</span></p>
      <p><span>Population:</span></p>
      <p><span>Time Zone:</span></p>
    </div>
  )
}

export const mapStateToProps = state => ({
  cityInfo: state.cityInfo
})

export default connect(mapStateToProps)(Comparison);