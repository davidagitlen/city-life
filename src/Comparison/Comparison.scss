.Comparison {
  display: grid;
  background-color: ghostwhite;
  height: 100vh;
  width: 100%;
  border: 1px solid green;
}