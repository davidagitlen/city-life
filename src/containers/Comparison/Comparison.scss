.comparison-chart__container {
  display: grid;
  grid-template: 1fr 1fr/1fr 1fr;
  background-color: ghostwhite;
  min-height: 112.3vh;
  width: 100%;
}

.instructions {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 250px;
    width: 470px;
    top: 250px;
    left: 250px;
    padding: 15px;
    font-size: 18px;
    background-color: rgba(46,46,201, .05)
  }

