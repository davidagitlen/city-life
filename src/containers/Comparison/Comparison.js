import React from 'react';
import './Comparison.scss';
import ComparisonChart from '../../components/ComparisonChart/ComparisonChart';
import { connect } from 'react-redux';
import { formatChartData } from '../../util/dataCleaner';

export const Comparison = (props) => {
  const { cityInfo } = props;
  const cityOneReady = cityInfo.one.scores.length && cityInfo.one.details && cityInfo.one.images.images;
  const cityTwoReady = cityInfo.two.scores.length && cityInfo.two.details && cityInfo.two.images.images;
  const cityOneName = cityOneReady ? cityInfo.one.details.fullName.split(',')[0] : null;
  const cityTwoName = cityTwoReady ? cityInfo.two.details.fullName.split(',')[0] : null;

  const cityOneScores = cityInfo.one.scores;
  const cityTwoScores = cityInfo.two.scores; 
  const economicData = formatChartData(cityOneScores, cityTwoScores, [0,1,2,3,6,11,12]);
  const healthAndSafetyData = formatChartData(cityOneScores, cityTwoScores, [7,8,10,16]);
  const culturalData = formatChartData(cityOneScores, cityTwoScores, [9,14,15]);
  const infrastructureData = formatChartData(cityOneScores, cityTwoScores, [4,5,13]);

  return(
    <div className='Comparison'>
      {(!cityOneReady || !cityTwoReady) && 
      <div name='instructions'>
      <p>Select two cities to from the form on the left to see a comparison of their quality of life data.</p>
      </div>}
      {(cityOneReady || null) && (cityTwoReady || null) && <ComparisonChart data={economicData} cityOneName={cityOneName} cityTwoName={cityTwoName}/>}
      {(cityOneReady || null) && (cityTwoReady || null) && <ComparisonChart data={healthAndSafetyData} cityOneName={cityOneName} cityTwoName={cityTwoName}/>}
      {(cityOneReady || null) && (cityTwoReady || null) && <ComparisonChart data={culturalData} cityOneName={cityOneName} cityTwoName={cityTwoName}/>}
      {(cityOneReady || null) && (cityTwoReady || null) && <ComparisonChart data={infrastructureData} cityOneName={cityOneName} cityTwoName={cityTwoName}/>}
    </div>
  )
}

export const mapStateToProps = state => ({
  cityInfo: state.cityInfo
})

export default connect(mapStateToProps)(Comparison);