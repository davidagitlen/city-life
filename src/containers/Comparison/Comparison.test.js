import React from 'react';
import { Comparison, mapStateToProps } from './Comparison';
import { shallow } from 'enzyme';

describe('Comparison', () => {
  let mockCityInfo;

  beforeEach(() => {
    mockCityInfo = {
      cityInfo: {
        one: {
          details: {
            population: 1,
            latitude: 10,
            longitude: 10,
            fullName: 'Denver, Colorado!',
            timeZone: 'Denver!'
          },
          images: {
            attribution: {},
            images: {
              mobile: 'http://blurg.com',
              web: 'http://blurg.com/web'
            }
          },
          scores: [{ color: 'blue', name: 'Score', score_out_of_10: 10 }]
        },
        two: {
          details: {
            population: 2,
            latitude: 20,
            longitude: 20,
            fullName: 'Boulder, Colorado!',
            timeZone: 'Denver!'
          },
          images: {
            attribution: {},
            images: {
              mobile: 'http://blurg.com',
              web: 'http://blurg.com/web'
            }
          },
          scores: [{ color: 'red', name: 'Score', score_out_of_10: 10 }]
        }
      }
    };
  })
  it('should match the snapshot with data passed in correctly', () => {

    const wrapper = shallow(
      <Comparison 
        {...mockCityInfo}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if some of the data is missing', () => {
    const mockCityInfo = {
      cityInfo: {
        one: {
          details: null,
          images: {
            attribution: {},
            images: {
              mobile: 'http://blurg.com',
              web: 'http://blurg.com/web'
            }
          },
          scores: [{ color: 'blue', name: 'Score', score_out_of_10: 10 }]
        },
        two: {
          details: null,
          images: {
            attribution: {},
            images: {
              mobile: 'http://blurg.com',
              web: 'http://blurg.com/web'
            }
          },
          scores: [{ color: 'red', name: 'Score', score_out_of_10: 10 }]
        }
      }
    };

    const wrapper = shallow(
      <Comparison 
        {...mockCityInfo}
      />
    );
    
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {

    it('should return an object with appropriate data', () => {

      const mockState = {
        cityInfo: mockCityInfo,
        unrealData: 'blarrrrgh!'
      };

      const expected = {
        cityInfo: mockCityInfo,
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

});