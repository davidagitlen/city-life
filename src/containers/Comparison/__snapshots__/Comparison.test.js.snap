// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison should match the snapshot if some of the data is missing 1`] = `
<div
  className="Comparison"
>
  <div
    name="instructions"
  >
    <p>
      Select two cities to from the form on the left to see a comparison of their quality of life data.
    </p>
  </div>
</div>
`;

exports[`Comparison should match the snapshot with data passed in correctly 1`] = `
<div
  className="Comparison"
>
  <ComparisonChart
    cityOneName="Denver"
    cityTwoName="Boulder"
    data={
      Object {
        "cityOneData": Array [
          Object {
            "color": "blue",
            "name": "Score",
            "score_out_of_10": 10,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "cityTwoData": Array [
          Object {
            "color": "red",
            "name": "Score",
            "score_out_of_10": 10,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
      }
    }
  />
  <ComparisonChart
    cityOneName="Denver"
    cityTwoName="Boulder"
    data={
      Object {
        "cityOneData": Array [
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "cityTwoData": Array [
          undefined,
          undefined,
          undefined,
          undefined,
        ],
      }
    }
  />
  <ComparisonChart
    cityOneName="Denver"
    cityTwoName="Boulder"
    data={
      Object {
        "cityOneData": Array [
          undefined,
          undefined,
          undefined,
        ],
        "cityTwoData": Array [
          undefined,
          undefined,
          undefined,
        ],
      }
    }
  />
  <ComparisonChart
    cityOneName="Denver"
    cityTwoName="Boulder"
    data={
      Object {
        "cityOneData": Array [
          undefined,
          undefined,
          undefined,
        ],
        "cityTwoData": Array [
          undefined,
          undefined,
          undefined,
        ],
      }
    }
  />
</div>
`;
